packages/core/src/dispatcher.ts

```diff
@@ -134,9 +141,22 @@ export default class Dispatcher extends EventEmitter {
    * @param cid - Record CID
    */
   async retrieveRecord (cid: CID | string): Promise<any> {
+    await this._restrictRecordSize(cid)
     return cloneDeep((await this._ipfs.dag.get(cid)).value)
```

Comment:
this won't work before we do `ipfs.dag.get` since we won't have the block in the blockstore yet.

Reply:
makes sense!

Reply:
does it make sense to get it and check it's size? I'm not sure if we can skip the `get` part 🤔

Reply:
yeah Don't think it's possible to check size without getting the data from the network.

Reply:
👍